tests/unit/form_tests/test_io_hdf5_h5tools.py

```diff
@@ -13,7 +13,9 @@ class H5IOTest(unittest.TestCase):
 
     def setUp(self):
         self.test_temp_file = tempfile.NamedTemporaryFile()
-        self.f = h5py.File(self.test_temp_file.name, 'w')
+        name = self.test_temp_file.name
+        self.test_temp_file.close()
```

Comment:
Generally speaking, this approach allow you to get the name of the file. But not that closing the handle delete the file.

This means that if the test suite is run from two processed on the same machine ... the same name may be allocated twice.

This is why I would recommend approach in https://github.com/scikit-build/scikit-ci/blob/9cae36620eddafec41d611c52a490e098cf7b7fa/ci/driver.py#L165-L166